python/ray/tests/kuberay/scripts/scale_down.py

```diff
@@ -1,12 +1,54 @@
 import ray
+from ray._private import test_utils
+import io
+import sys
+
+SCALE_DOWN_CPU = "Removing 1 nodes of type fake-gpu-group (idle)."
+SCALE_DOWN_GPU = "Removing 2 nodes of type small-group (idle)."
 
 
 def main():
-    """Removes CPU request, removes GPU actor."""
+    """Removes CPU request, removes GPU actor.
+    Waits for autoscaler scale-down events to get emitted to stdout."""
+    # Redirect stdout to a StringIO object.
+    out_stream = io.StringIO()
+    sys.stdout = out_stream
+
+    # Remove resource demands
     ray.autoscaler.sdk.request_resources(num_cpus=0)
     gpu_actor = ray.get_actor("gpu_actor")
     ray.kill(gpu_actor)
 
+    def scale_events_logged() -> bool:
+        """Return True if the scale down event messages are in the out_stream.
+        Otherwise, raise an Exception.
+        """
+        stdout_so_far = out_stream.getvalue()
+        if (SCALE_DOWN_CPU in stdout_so_far) and (SCALE_DOWN_GPU in stdout_so_far):
+            return True
+        else:
+            # This error will be logged to stderr once if SCALE_MESSAGE doesn't
+            # appear in STDOUT in time.
+            raise RuntimeError(
```

Comment:
The issue is that wait for condition would throw without displaying an error message.

The solution is to refactor wait for condition a bit (rather than attempting to work around its current structure).

Reply:
Took care of logging to stdout in a finally clause in the new decorator.
The condition is now a True/False lambda.